src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs

```diff
@@ -91,7 +91,10 @@ public void Reanalyze(Workspace workspace, IIncrementalAnalyzer analyzer, IEnume
                 var coordinator = default(WorkCoordinator);
                 if (!_documentWorkCoordinatorMap.TryGetValue(workspace, out coordinator))
                 {
-                    throw new ArgumentException("workspace");
+                    // this can happen if solution crawler is already unregistered from workspace.
+                    // one of those example will be VS shutting down so roslyn package is disposed but there is a pending
+                    // async operation.
```

Comment:
here are what main thread is doing.

msenv!CVsPackageMgr::ClosePackageInfo+0x7
msenv!<lambda_6291f8f33a08c6cf7e2926ec2f4b5c33>::operator()+0xcc
msenv!std::_Func_class<long,std::shared_ptr<CVsPackageMgr::CVsPackageInfo> const &>::operator()+0x14
msenv!CVsPackageMgr::ThreadSafePackageCollection::IteratePackages+0x146
msenv!CVsPackageMgr::ThreadSafePackageCollection::IteratePackages+0x45
msenv!CVsPackageMgr::CloseAllPackages+0x14e
msenv!InitTermThread+0x90
msenv!InitTerminate+0xd6
msenv!VStudioMainLogged+0x613
msenv!VStudioMain+0x7c
devenv!util_CallVsMain+0xde
devenv!CDevEnvAppId::Run+0xb99
devenv!WinMain+0xb8
devenv!invoke_main+0x1a